CodePush.ios.js

```diff
@@ -95,12 +96,86 @@ function notifyApplicationReady() {
   return NativeCodePush.notifyApplicationReady();
 }
 
+/**
+ * The sync method provides a simple, one-line experience for
+ * incorporating the check, download and application of an update.
+ * 
+ * It simply composes the existing API methods together and adds additional
+ * support for respecting mandatory updates, ignoring previously failed
+ * releases, and displaying a standard confirmation UI to the end-user
+ * when an update is available.
+ */
+function sync(options = {}) {  
+  var syncOptions = {
+    ignoreFailedUpdates: true,
+    
+    mandatoryContinueButtonLabel: "Continue",
+    mandatoryUpdateMessage: "An update is available that must be installed",
+    
+    optionalIgnoreButtonLabel: "Ignore",
+    optionalInstallButtonLabel: "Install",
+    optionalUpdateMessage: "An update is available. Would you like to install it?",
+    
+    updateTitle: "Update available",
+    rollbackTimeout: 0,
+    
+    ...options 
+  };
+
+  return new Promise((resolve, reject) => {
+    checkForUpdate()
+      .then((remotePackage) => {
+        if (!remotePackage || (remotePackage.failedAppy && syncOptions.ignoreFailedUpdates)) {
+          resolve(CodePush.SyncStatus.NO_UPDATE_AVAILABLE);
+        }
+        else {
+          var message = null;
+          var dialogButtons = [
+            {
+              text: null,
+              onPress: () => { 
+                remotePackage.download()
+                  .then((localPackage) => {
+                    resolve(CodePush.SyncStatus.APPLY_SUCCESS);
```

Comment:
I agree this name isn't great. APPLY_SUCCESS is also being used for the Cordova plugin, so I'm cool with a name change as long as we're consistent.